walletkit_client.go

```diff
@@ -424,6 +434,153 @@ func (m *walletKitClient) ListSweeps(ctx context.Context) ([]string, error) {
+
+	case lnrpc.OutputScriptType_SCRIPT_TYPE_WITNESS_UNKNOWN:
+		return txscript.WitnessUnknownTy
+
+	case lnrpc.OutputScriptType_SCRIPT_TYPE_WITNESS_V1_TAPROOT:
+		return txscript.WitnessV1TaprootTy
+
+	default:
+		return txscript.NonStandardTy
+	}
+}
+
+// RPCTransaction returns a rpc transaction.
+func UnmarshalTransactionDetail(tx *lnrpc.Transaction,
+	chainParams *chaincfg.Params) (*lnwallet.TransactionDetail, error) {
+
+	var outputDetails []lnwallet.OutputDetail
+	for _, o := range tx.OutputDetails {
+		address, err := btcutil.DecodeAddress(o.Address, chainParams)
+		if err != nil {
+			return nil, err
+		}
+
+		pkScript, err := hex.DecodeString(o.PkScript)
+		if err != nil {
+			return nil, err
+		}
+
+		outputDetails = append(outputDetails, lnwallet.OutputDetail{
+			OutputType:   unmarshallOutputType(o.OutputType),
+			Addresses:    []btcutil.Address{address},
+			PkScript:     pkScript,
+			OutputIndex:  int(o.OutputIndex),
+			Value:        btcutil.Amount(o.Amount),
+			IsOurAddress: o.IsOurAddress,
+		})
+	}
+
+	previousOutpoints := make(
+		[]lnwallet.PreviousOutPoint, len(tx.PreviousOutpoints),
+	)
+	for idx, previousOutPoint := range tx.PreviousOutpoints {
+		previousOutpoints[idx] = lnwallet.PreviousOutPoint{
+			OutPoint:    previousOutPoint.Outpoint,
+			IsOurOutput: previousOutPoint.IsOurOutput,
+		}
+	}
+
+	// We also get unconfirmed transactions, so BlockHash can be empty.
+	var (
+		blockHash *chainhash.Hash
+		err       error
+	)
+
+	if tx.BlockHash != "" {
+		blockHash, err = chainhash.NewHashFromStr(tx.BlockHash)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	txHash, err := chainhash.NewHashFromStr(tx.TxHash)
+	if err != nil {
+		return nil, err
+	}
+
+	rawTx, err := hex.DecodeString(tx.RawTxHex)
+	if err != nil {
+		return nil, err
+	}
+
+	return &lnwallet.TransactionDetail{
+		Hash:              *txHash,
+		Value:             btcutil.Amount(tx.Amount),
+		NumConfirmations:  tx.NumConfirmations,
+		BlockHash:         blockHash,
+		BlockHeight:       tx.BlockHeight,
+		Timestamp:         tx.TimeStamp,
+		TotalFees:         tx.TotalFees,
+		OutputDetails:     outputDetails,
+		RawTx:             rawTx,
+		Label:             tx.Label,
+		PreviousOutpoints: previousOutpoints,
+	}, nil
+}
+
+// ListSweepsVerbose returns a list of sweep transactions known to our node
+// with verbose information about each sweep.
+func (m *walletKitClient) ListSweepsVerbose(ctx context.Context) (
+	[]lnwallet.TransactionDetail, error) {
+
+	rpcCtx, cancel := context.WithTimeout(ctx, m.timeout)
+	defer cancel()
+
+	resp, err := m.client.ListSweeps(
+		m.walletKitMac.WithMacaroonAuth(rpcCtx),
+		&walletrpc.ListSweepsRequest{
+			Verbose: true,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	var result []lnwallet.TransactionDetail
+	// Since we have requested the abbreviated response from lnd, we can
+	// just get our response to a list of sweeps and return it.
+	rpcDetails := resp.GetTransactionDetails()
+	if rpcDetails != nil {
```

Comment:
Should we error out if this is nil? Since we explicitly requested the verbose version.